packages/react/src/components/MenuButton/index.tsx

```diff
@@ -178,7 +218,42 @@ const MenuButton = forwardRef<HTMLDivElement, MenuButtonProps>(
           refs.floating.current.style[style] = value;
         }
       });
-    }, [floatingStyles, refs.floating, middlewareData, placement, open]);
+
+      if (refs.floating.current && middlewareData.hide) {
+        refs.floating.current.style.visibility = middlewareData.hide
+          .referenceHidden
+          ? 'hidden'
+          : 'visible';
+      }
+
+      if (refs.floating.current && placement) {
+        const placementClasses = [
+          'top',
+          'bottom',
+          'left',
+          'right',
+          'top-start',
+          'top-end',
+          'bottom-start',
+          'bottom-end',
+          'left-start',
+          'left-end',
+          'right-start',
+          'right-end',
+        ].map((p) => `${prefix}--menu--${p}`);
+
+        refs.floating.current.classList.remove(...placementClasses);
+
+        refs.floating.current.classList.add(`${prefix}--menu--${placement}`);
+      }
+    }, [
+      floatingStyles,
+      refs.floating,
+      middlewareData,
+      placement,
+      open,
+      prefix,
+    ]);
```

Comment:
This isn't needed, all styles are applied via CSSOM instead of by classnames.

https://github.com/carbon-design-system/carbon/blob/20b4691222ebde34ae61de44cd3e029e4a4dc42d/packages/react/src/components/MenuButton/index.tsx#L167-L180